services/src/handlers/datasets.rs

```diff
@@ -134,4 +198,71 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn create_data_set() {
+        let ctx = InMemoryContext::default();
+
+        let session = create_session_helper(&ctx).await;
+
+        let s = r#"{
+            "upload": "1f7e3e75-4d20-4c91-9497-7f4df7604b62",
+            "definition": {
+                "properties": {
+                    "id": null,
```

Comment:
good point. I think we need a seperate DTO-struct for the annotation of upload data and not use the MetaDataDefinition directly. We need a guided process of specifying this anyway (e.g. csv-importer) and ideally some auto-suggestions (if I upload a CSV it should ask if I want to specify columns, if I upload a tiff I need it can use gdalinfo to suggest data type etc.

I think it's a separate task to think about/do this stuff.

Reply:
Okay, but what happens now? Does it throw an error? Undefined behavior? Panic?

Reply:
the in-memory data set db would overwrite the dataset. Maybe it is not bad to have such upsert behavior - provided the user is owner of the specified data set. Currently this is not checked though.